src/legacy/ui/public/new_platform/new_platform.ts

```diff
@@ -68,6 +70,21 @@ export function __reset__() {
 export function __setup__(coreSetup: LegacyCoreSetup, plugins: PluginsSetup) {
   npSetup.core = coreSetup;
   npSetup.plugins = plugins;
+
+  // Setup compatibility layer for AppService in legacy platform
+  npSetup.core.application.register = (app: App) => {
```

Comment:
+1 for a functional test. we don't have too many of them for integration points, but it would be a good example how to use this API in the legacy platform